### ISSUE TITLE
Option to print out files parsed

### ISSUE THREAD

**Issue by ElvenSpellmaker** [adrienverge/yamllint#396]
Hey, I've been using YAMLLint for some time, but it'd be nice to print out which files are detected and have been parsed for linting.

This is useful to check the right files are being picked up, we had a situation where one `yamllint.conf` file wasn't set up right and so no files were being linted but in the builds we didn't detect it for some time as the file list found is never printed.

Feature: A flag to show the parsed files, and optionally an error code if _no_ files have been parsed for easier build detection.

**Comment by adrienverge** [comment]
Hello, thanks for the idea.

The behavior you propose doesn't seem consistent with most standard tools (grep, sed, find, git...), so I'm not very favorable to add an option for this.

If you really need to know which files are read, you could either make sure `.yamllint` is valid, or explicitly pass these files to yamllint:
```shell
find -regex '.*\.ya?ml'
yamllint $(find -regex '.*\.ya?ml')
yamllint $(any other command you like)
```
```

**Comment by splattael** [comment]
@adrienverge While you can use tools to determine which files are being parsed by yamllint it gets trickier if [ignoring paths](https://yamllint.readthedocs.io/en/stable/configuration.html#ignoring-paths) comes into play.

In https://gitlab.com/gitlab-org/gitlab/-/merge_requests/107001 I used `strace -e trace=openat` to track files opened by yamllint. Not very convenient but it helps to detect which files are being opened before and after a configuration change.

Still, I wonder if we could add command line switch `--list-files` to just print the files and exit. It's actually very similar to what [RuboCop is doing with `--list-targets`](https://docs.rubocop.org/rubocop/usage/basic_usage.html#command-line-flags).

I'll shoot a PR and reference this issue :muscle:

**Comment by splattael** [comment]
See https://github.com/adrienverge/yamllint/pull/524
